.github/workflows/ci.yml

```diff
@@ -11,34 +11,37 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: "3.9"
-      - uses: Gr1n/setup-poetry@v4
+          python-version: "3.10"
+      - uses: Gr1n/setup-poetry@v7
         with:
-          poetry-version: '1.1.4'
+          poetry-version: '1.1.12'
       - run: pip install tox
-      - run: tox -e lint,py39-dj32
+      - run: tox -e lint,py310-dj32
   test_compatibility:
     needs: test
     runs-on: ubuntu-latest
     strategy:
       matrix:
         include:
-          - python: "3.6"
-            toxenv: py36-dj22,py36-dj30,py36-dj31,py36-dj32
+          # Test with all supported Django versions, for all compatible Python versions.
+          # See https://docs.djangoproject.com/en/4.0/faq/install/#what-python-version-can-i-use-with-django for the official matrix.
+          # Additionally test on Django’s main branch with the most recent Python version.
           - python: "3.7"
             toxenv: py37-dj22,py37-dj30,py37-dj31,py37-dj32
           - python: "3.8"
-            toxenv: py38-dj22,py38-dj30,py38-dj31,py38-dj32,py38-djmain
+            toxenv: py38-dj22,py38-dj30,py38-dj31,py38-dj32
           - python: "3.9"
-            toxenv: py39-dj22,py39-dj30,py39-dj31,py39-dj32,py39-djmain
+            toxenv: py39-dj22,py39-dj30,py39-dj31,py39-dj32
+          - python: "3.10"
+            toxenv: py310-dj32,py310-djmain
```

Comment:
Two things worth mentioning here:

- I thought it would be enough to only test `djmain` with the most recent Python version
- Similarly since `py31-dj32` is already tested in the previous `test` job, I’ve removed it here.